File: Data Structures & Algorithms/maximum-subarray/submission-3.go
func maxSubArray(nums []int) int {
    currsum := nums[0]
    maxsum := nums[0]
    n := len(nums)

    for i:=1;i<n;i++{
        if currsum+nums[i] > nums[i]{
            currsum = currsum + nums[i]
        }else{
            currsum = nums[i]
        }
        if currsum > maxsum{
            maxsum = currsum
        }
    }
    return maxsum
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-3.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int cnt = 0;
        for(int i = 0; i<nums.size();i++){
            int j = i;
            int currsum = 0;
            while(j<nums.size()){
                currsum += nums[j];
                if(currsum == k){
                    cnt++;
                }
                j++;
            }
        }

        return cnt;
    }
};